pynicotine/gtkgui/userbrowse.py

```diff
@@ -357,8 +439,22 @@ def on_expand(self, widget):
             self.expand.set_property("icon-name", "go-down-symbolic")
 
     def on_folder_row_activated(self, treeview, path, column):
-        if self.user != config.sections["server"]["login"]:
-            self.on_download_directory()
+
+        if path is None:
+            return
+
+        # Keyboard accessibility support for <Return> key behaviour
+        if self.FolderTreeView.row_expanded(path):
+            expandable = self.FolderTreeView.collapse_row(path)
+        else:
+            expandable = self.FolderTreeView.expand_row(path, False)
+
+        if not expandable and len(self.file_store) > 0:
+            # This is the deepest level, so move focus over to Files if there are any
+            self.FileTreeView.grab_focus()
+
+        # Note: Other Folder actions are handled by setup_accelerator functions [Shift/Ctrl/Alt+Return]
+        # ToDo: Mouse double-click actions will need *args for keycode state & mods [Ctrl/Alt+DblClick]         # ToDo #
 
```

Comment:
Now that the accelerators each have their own function definition, this will allow selective Mouse-click events to be assigned to them, this is why I have specified the word "Enter" (rather than "Return") in the code comments below.

_See link: https://anzeljg.github.io/rin2/book2/2405/docs/tkinter/key-names.html_
`Return | 36 | 65293 | The enter key... The name Enter refers to a mouse-related event, not a keypress;`